names/names.py

```diff
@@ -11,12 +40,24 @@
 f.close()
 
 duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# for name_1 in names_1:
+#     for name_2 in names_2:
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+
+
+b = BinarySearchTree('')
+for i in names_1:
+  b.insert(i)
+  
+
+for j in names_2:
+  if b.contains(j):
+    duplicates.append(j)
+
```

Comment:
Nice use of BST to get a well optimized nlog(n) solution in terms of time complexity.  Consider using a set or hash-table to solve something like this in the future.  Since they have O(1) look-up as opposed to BST's O(log(n)), the overall time complexity of the solution could be O(n) instead of O(nlog(n)).